imap/jmap_mail_submission.c

```diff
@@ -909,15 +909,11 @@ static void _emailsubmission_update(struct mailbox *submbox,
                     }
                 }
                 else if (!strcmp(arg, "undoStatus")) {
-                    if (record->internal_flags & FLAG_INTERNAL_EXPUNGED) {
-                        if (!(record->internal_flags & FLAG_INTERNAL_UNLINKED)) {
-                            /* Already sent */
-                            *set_err = json_pack("{s:s}", "type", "cannotUnsend");
-                        }
-                        else if (!strcmp(strval, "canceled")) {
-                            /* Already canceled */
-                            continue;
-                        }
+                    if (record->system_flags & FLAG_ANSWERED) {
+                        if (!strcmp(strval, "final")) continue;
```

Comment:
The cannotUnsend error is missing for the "final" case

Reply:
hmm yeah good point.

Reply:
Already fixed after merge.